--- SWM050-patched.svd.orig	2019-04-16 13:36:27.287488205 +0200
+++ SWM050-patched.svd	2019-04-26 12:57:53.589758709 +0200
@@ -412,6 +412,46 @@
         <size>0x00000800</size>
         <usage>registers</usage>
       </addressBlock>
+      <interrupt>
+        <name>GPIOA0</name>
+        <value>4</value>
+      </interrupt>
+      <interrupt>
+        <name>GPIOA1</name>
+        <value>5</value>
+      </interrupt>
+      <interrupt>
+        <name>GPIOA2</name>
+        <value>6</value>
+      </interrupt>
+      <interrupt>
+        <name>GPIOA3</name>
+        <value>7</value>
+      </interrupt>
+      <interrupt>
+        <name>GPIOA4</name>
+        <value>8</value>
+      </interrupt>
+      <interrupt>
+        <name>GPIOA5</name>
+        <value>9</value>
+      </interrupt>
+      <interrupt>
+        <name>GPIOA6</name>
+        <value>10</value>
+      </interrupt>
+      <interrupt>
+        <name>GPIOA7</name>
+        <value>11</value>
+      </interrupt>
+      <interrupt>
+        <name>GPIOA8</name>
+        <value>12</value>
+      </interrupt>
+      <interrupt>
+        <name>GPIOA9</name>
+        <value>12</value>
+      </interrupt>
       <registers>
         <register>
           <name>DAT</name>
@@ -1225,6 +1265,10 @@
         <size>0x00000800</size>
         <usage>registers</usage>
       </addressBlock>
+      <interrupt>
+        <name>ACMP</name>
+        <value>3</value>
+      </interrupt>
       <registers>
         <register>
           <name>CPEN</name>
@@ -1610,6 +1654,10 @@
         <size>0x00000800</size>
         <usage>registers</usage>
       </addressBlock>
+      <interrupt>
+        <name>WDT</name>
+        <value>2</value>
+      </interrupt>
       <registers>
         <register>
           <name>CR</name>
@@ -1707,6 +1755,10 @@
         <size>0x00000800</size>
         <usage>registers</usage>
       </addressBlock>
+      <interrupt>
+        <name>TMRSE0</name>
+        <value>0</value>
+      </interrupt>
       <registers>
         <register>
           <name>CTRL</name>
@@ -1906,6 +1958,10 @@
         <size>0x00000800</size>
         <usage>registers</usage>
       </addressBlock>
+      <interrupt>
+        <name>TMRSE1</name>
+        <value>1</value>
+      </interrupt>
     </peripheral>
   </peripherals>
 </device>
