--- SWM050-patched.svd.orig	2019-05-03 14:10:56.139767129 +0200
+++ SWM050-patched.svd	2019-05-03 15:38:48.772872900 +0200
@@ -1728,12 +1728,26 @@
         </register>
         <register>
           <name>CRR</name>
-          <description>CRR</description>
+          <description>Counter Restart Register</description>
           <addressOffset>0x0000000C</addressOffset>
-          <access>read-only</access>
+          <access>write-only</access>
           <resetValue>0x00000000</resetValue>
           <resetMask>0xFFFFFFFF</resetMask>
-          <fields></fields>
+          <fields>
+            <field>
+              <name>CRR</name>
+              <bitOffset>0</bitOffset>
+              <bitWidth>8</bitWidth>
+              <enumeratedValues>
+                <name>CRR</name>
+                <enumeratedValue>
+                  <name>Reset</name>
+                  <description>Reset WDT counter</description>
+                  <value>118</value>
+                </enumeratedValue>
+              </enumeratedValues>
+            </field>
+          </fields>
         </register>
         <register>
           <name>STAT</name>
