--- SWM050.svd.orig	2019-04-01 22:07:46.571075624 +0200
+++ SWM050.svd	2019-04-08 10:27:13.025703394 +0200
@@ -567,571 +567,400 @@
             </field>
           </fields>
         </register>
-      </registers>
-    </peripheral>
-    <peripheral>
-      <name>GPIO_EXTA</name>
-      <description>Registers group</description>
-      <groupName>GPIO_EXTA</groupName>
-      <baseAddress>0x40001050</baseAddress>
-      <addressBlock>
-        <offset>0x00000000</offset>
-        <size>0x00000800</size>
-        <usage>registers</usage>
-      </addressBlock>
-      <registers>
         <register>
-          <name>EXT</name>
-          <description>EXT</description>
-          <addressOffset>0x00000000</addressOffset>
-          <access>read-only</access>
+          <name>INTEN</name>
+          <description>INTEN</description>
+          <addressOffset>0x00000030</addressOffset>
+          <access>read-write</access>
           <resetValue>0x00000000</resetValue>
           <resetMask>0xFFFFFFFF</resetMask>
           <fields>
             <field>
-              <name>EXT0</name>
-              <description>EXT0</description>
+              <name>INTEN0</name>
+              <description>INTEN0</description>
               <bitOffset>0</bitOffset>
               <bitWidth>1</bitWidth>
               <access>read-write</access>
             </field>
             <field>
-              <name>EXT1</name>
-              <description>EXT1</description>
+              <name>INTEN1</name>
+              <description>INTEN1</description>
               <bitOffset>1</bitOffset>
               <bitWidth>1</bitWidth>
               <access>read-write</access>
             </field>
             <field>
-              <name>EXT2</name>
-              <description>EXT2</description>
+              <name>INTEN2</name>
+              <description>INTEN2</description>
               <bitOffset>2</bitOffset>
               <bitWidth>1</bitWidth>
               <access>read-write</access>
             </field>
             <field>
-              <name>EXT3</name>
-              <description>EXT3</description>
+              <name>INTEN3</name>
+              <description>INTEN3</description>
               <bitOffset>3</bitOffset>
               <bitWidth>1</bitWidth>
               <access>read-write</access>
             </field>
             <field>
-              <name>EXT4</name>
-              <description>EXT4</description>
+              <name>INTEN4</name>
+              <description>INTEN4</description>
               <bitOffset>4</bitOffset>
               <bitWidth>1</bitWidth>
               <access>read-write</access>
             </field>
             <field>
-              <name>EXT5</name>
-              <description>EXT5</description>
+              <name>INTEN5</name>
+              <description>INTEN5</description>
               <bitOffset>5</bitOffset>
               <bitWidth>1</bitWidth>
               <access>read-write</access>
             </field>
             <field>
-              <name>EXT6</name>
-              <description>EXT6</description>
+              <name>INTEN6</name>
+              <description>INTEN6</description>
               <bitOffset>6</bitOffset>
               <bitWidth>1</bitWidth>
               <access>read-write</access>
             </field>
             <field>
-              <name>EXT7</name>
-              <description>EXT7</description>
+              <name>INTEN7</name>
+              <description>INTEN7</description>
               <bitOffset>7</bitOffset>
               <bitWidth>1</bitWidth>
               <access>read-write</access>
             </field>
             <field>
-              <name>EXT8</name>
-              <description>EXT8</description>
+              <name>INTEN8</name>
+              <description>INTEN8</description>
               <bitOffset>8</bitOffset>
               <bitWidth>1</bitWidth>
               <access>read-write</access>
             </field>
             <field>
-              <name>EXT9</name>
-              <description>EXT9</description>
+              <name>INTEN9</name>
+              <description>INTEN9</description>
               <bitOffset>9</bitOffset>
               <bitWidth>1</bitWidth>
               <access>read-write</access>
             </field>
           </fields>
         </register>
-      </registers>
-    </peripheral>
-    <peripheral>
-      <name>ACMP</name>
-      <description>Registers group</description>
-      <groupName>ACMP</groupName>
-      <baseAddress>0x40000800</baseAddress>
-      <addressBlock>
-        <offset>0x00000000</offset>
-        <size>0x00000800</size>
-        <usage>registers</usage>
-      </addressBlock>
-      <registers>
-        <register>
-          <name>CPEN</name>
-          <description>CPEN</description>
-          <addressOffset>0x00000000</addressOffset>
-          <access>read-write</access>
-          <resetValue>0x00000000</resetValue>
-          <resetMask>0xFFFFFFFF</resetMask>
-          <fields>
-            <field>
-              <name>EN</name>
-              <description>EN</description>
-              <bitOffset>4</bitOffset>
-              <bitWidth>1</bitWidth>
-              <access>read-write</access>
-            </field>
-          </fields>
-        </register>
-        <register>
-          <name>CPLEVEL</name>
-          <description>CPLEVEL</description>
-          <addressOffset>0x00000004</addressOffset>
-          <access>read-write</access>
-          <resetValue>0x00000000</resetValue>
-          <resetMask>0xFFFFFFFF</resetMask>
-          <fields>
-            <field>
-              <name>LEVEL</name>
-              <description>LEVEL</description>
-              <bitOffset>0</bitOffset>
-              <bitWidth>8</bitWidth>
-              <access>read-write</access>
-            </field>
-          </fields>
-        </register>
         <register>
-          <name>CPOUTPUT</name>
-          <description>CPOUTPUT</description>
-          <addressOffset>0x00000010</addressOffset>
+          <name>INTMSK</name>
+          <description>INTMSK</description>
+          <addressOffset>0x00000034</addressOffset>
           <access>read-write</access>
           <resetValue>0x00000000</resetValue>
           <resetMask>0xFFFFFFFF</resetMask>
           <fields>
             <field>
-              <name>CPOUTRAW</name>
-              <description>CPOUTRAW</description>
+              <name>INTMSK0</name>
+              <description>INTMSK0</description>
               <bitOffset>0</bitOffset>
               <bitWidth>1</bitWidth>
               <access>read-write</access>
             </field>
             <field>
-              <name>CPOUTMOS</name>
-              <description>CPOUTMOS</description>
+              <name>INTMSK1</name>
+              <description>INTMSK1</description>
               <bitOffset>1</bitOffset>
               <bitWidth>1</bitWidth>
               <access>read-write</access>
             </field>
             <field>
-              <name>CPOUTCEL</name>
-              <description>CPOUTCEL</description>
+              <name>INTMSK2</name>
+              <description>INTMSK2</description>
               <bitOffset>2</bitOffset>
               <bitWidth>1</bitWidth>
               <access>read-write</access>
             </field>
-          </fields>
-        </register>
-        <register>
-          <name>CPMOD</name>
-          <description>CPMOD</description>
-          <addressOffset>0x00000020</addressOffset>
-          <access>read-write</access>
-          <resetValue>0x00000000</resetValue>
-          <resetMask>0xFFFFFFFF</resetMask>
-          <fields>
             <field>
-              <name>WORKSMOD</name>
-              <description>WORKSMOD</description>
-              <bitOffset>0</bitOffset>
+              <name>INTMSK3</name>
+              <description>INTMSK3</description>
+              <bitOffset>3</bitOffset>
               <bitWidth>1</bitWidth>
               <access>read-write</access>
             </field>
-          </fields>
-        </register>
-        <register>
-          <name>CPCTL</name>
-          <description>CPCTL</description>
-          <addressOffset>0x00000024</addressOffset>
-          <access>read-write</access>
-          <resetValue>0x00000000</resetValue>
-          <resetMask>0xFFFFFFFF</resetMask>
-          <fields>
             <field>
-              <name>MOS_LEVEL</name>
-              <description>MOS_LEVEL</description>
-              <bitOffset>1</bitOffset>
+              <name>INTMSK4</name>
+              <description>INTMSK4</description>
+              <bitOffset>4</bitOffset>
               <bitWidth>1</bitWidth>
               <access>read-write</access>
             </field>
             <field>
-              <name>CEL_LEVEL</name>
-              <description>CEL_LEVEL</description>
-              <bitOffset>2</bitOffset>
+              <name>INTMSK5</name>
+              <description>INTMSK5</description>
+              <bitOffset>5</bitOffset>
               <bitWidth>1</bitWidth>
               <access>read-write</access>
             </field>
             <field>
-              <name>A7M_LEVEL</name>
-              <description>A7M_LEVEL</description>
-              <bitOffset>3</bitOffset>
+              <name>INTMSK6</name>
+              <description>INTMSK6</description>
+              <bitOffset>6</bitOffset>
               <bitWidth>1</bitWidth>
               <access>read-write</access>
             </field>
             <field>
-              <name>A7C_LEVEL</name>
-              <description>A7C_LEVEL</description>
-              <bitOffset>4</bitOffset>
+              <name>INTMSK7</name>
+              <description>INTMSK7</description>
+              <bitOffset>7</bitOffset>
               <bitWidth>1</bitWidth>
               <access>read-write</access>
             </field>
             <field>
-              <name>MOS_DELAY</name>
-              <description>MOS_DELAY</description>
+              <name>INTMSK8</name>
+              <description>INTMSK8</description>
               <bitOffset>8</bitOffset>
-              <bitWidth>6</bitWidth>
-              <access>read-write</access>
-            </field>
-            <field>
-              <name>CEL_DELAY</name>
-              <description>CEL_DELAY</description>
-              <bitOffset>16</bitOffset>
-              <bitWidth>6</bitWidth>
+              <bitWidth>1</bitWidth>
               <access>read-write</access>
             </field>
-          </fields>
-        </register>
-        <register>
-          <name>CLRLOCK</name>
-          <description>CLRLOCK</description>
-          <addressOffset>0x00000028</addressOffset>
-          <access>read-write</access>
-          <resetValue>0x00000000</resetValue>
-          <resetMask>0xFFFFFFFF</resetMask>
-          <fields>
             <field>
-              <name>CLR</name>
-              <description>CLR</description>
-              <bitOffset>0</bitOffset>
+              <name>INTMSK9</name>
+              <description>INTMSK9</description>
+              <bitOffset>9</bitOffset>
               <bitWidth>1</bitWidth>
               <access>read-write</access>
             </field>
           </fields>
         </register>
         <register>
-          <name>LOCKCON</name>
-          <description>LOCKCON</description>
-          <addressOffset>0x0000002C</addressOffset>
+          <name>INTMODE</name>
+          <description>INTMODE</description>
+          <addressOffset>0x00000038</addressOffset>
           <access>read-write</access>
           <resetValue>0x00000000</resetValue>
           <resetMask>0xFFFFFFFF</resetMask>
           <fields>
             <field>
-              <name>LOCKEN</name>
-              <description>LOCKEN</description>
+              <name>INTMODE0</name>
+              <description>INTMODE0</description>
               <bitOffset>0</bitOffset>
               <bitWidth>1</bitWidth>
               <access>read-write</access>
             </field>
             <field>
-              <name>LEVELCON</name>
-              <description>LEVELCON</description>
+              <name>INTMODE1</name>
+              <description>INTMODE1</description>
               <bitOffset>1</bitOffset>
               <bitWidth>1</bitWidth>
               <access>read-write</access>
             </field>
             <field>
-              <name>CLRCON</name>
-              <description>CLRCON</description>
-              <bitOffset>2</bitOffset>
-              <bitWidth>1</bitWidth>
-              <access>read-write</access>
-            </field>
-            <field>
-              <name>LOCKSTAT</name>
-              <description>LOCKSTAT</description>
-              <bitOffset>3</bitOffset>
-              <bitWidth>1</bitWidth>
-              <access>read-write</access>
-            </field>
-          </fields>
-        </register>
-        <register>
-          <name>CPINTCON</name>
-          <description>CPINTCON</description>
-          <addressOffset>0x00000040</addressOffset>
-          <access>read-write</access>
-          <resetValue>0x00000000</resetValue>
-          <resetMask>0xFFFFFFFF</resetMask>
-          <fields>
-            <field>
-              <name>INTMASKn</name>
-              <description>INTMASKn</description>
-              <bitOffset>0</bitOffset>
-              <bitWidth>1</bitWidth>
-              <access>read-write</access>
-            </field>
-            <field>
-              <name>INTUP</name>
-              <description>INTUP</description>
+              <name>INTMODE2</name>
+              <description>INTMODE2</description>
               <bitOffset>2</bitOffset>
               <bitWidth>1</bitWidth>
               <access>read-write</access>
             </field>
             <field>
-              <name>INTDOWN</name>
-              <description>INTDOWN</description>
+              <name>INTMODE3</name>
+              <description>INTMODE3</description>
               <bitOffset>3</bitOffset>
               <bitWidth>1</bitWidth>
               <access>read-write</access>
             </field>
             <field>
-              <name>MOSINTMASKn</name>
-              <description>MOSINTMASKn</description>
+              <name>INTMODE4</name>
+              <description>INTMODE4</description>
               <bitOffset>4</bitOffset>
               <bitWidth>1</bitWidth>
               <access>read-write</access>
             </field>
             <field>
-              <name>MOSINTEN</name>
-              <description>MOSINTEN</description>
+              <name>INTMODE5</name>
+              <description>INTMODE5</description>
               <bitOffset>5</bitOffset>
               <bitWidth>1</bitWidth>
               <access>read-write</access>
             </field>
             <field>
-              <name>MOSINTUP</name>
-              <description>MOSINTUP</description>
+              <name>INTMODE6</name>
+              <description>INTMODE6</description>
               <bitOffset>6</bitOffset>
               <bitWidth>1</bitWidth>
               <access>read-write</access>
             </field>
             <field>
-              <name>MOSINTDOWN</name>
-              <description>MOSINTDOWN</description>
+              <name>INTMODE7</name>
+              <description>INTMODE7</description>
               <bitOffset>7</bitOffset>
               <bitWidth>1</bitWidth>
               <access>read-write</access>
             </field>
-          </fields>
-        </register>
-        <register>
-          <name>CPINTRAW</name>
-          <description>CPINTRAW</description>
-          <addressOffset>0x00000044</addressOffset>
-          <access>read-write</access>
-          <resetValue>0x00000000</resetValue>
-          <resetMask>0xFFFFFFFF</resetMask>
-          <fields>
-            <field>
-              <name>CPRAW</name>
-              <description>CPRAW</description>
-              <bitOffset>0</bitOffset>
-              <bitWidth>1</bitWidth>
-              <access>read-write</access>
-            </field>
             <field>
-              <name>CPMOS</name>
-              <description>CPMOS</description>
-              <bitOffset>1</bitOffset>
+              <name>INTMODE8</name>
+              <description>INTMODE8</description>
+              <bitOffset>8</bitOffset>
               <bitWidth>1</bitWidth>
               <access>read-write</access>
             </field>
             <field>
-              <name>CPCEL</name>
-              <description>CPCEL</description>
-              <bitOffset>2</bitOffset>
+              <name>INTMODE9</name>
+              <description>INTMODE9</description>
+              <bitOffset>9</bitOffset>
               <bitWidth>1</bitWidth>
               <access>read-write</access>
             </field>
           </fields>
         </register>
         <register>
-          <name>CPINT</name>
-          <description>CPINT</description>
-          <addressOffset>0x00000048</addressOffset>
+          <name>INTLEVEL</name>
+          <description>INTLEVEL</description>
+          <addressOffset>0x0000003C</addressOffset>
           <access>read-write</access>
           <resetValue>0x00000000</resetValue>
           <resetMask>0xFFFFFFFF</resetMask>
           <fields>
             <field>
-              <name>CPRAW</name>
-              <description>CPRAW</description>
+              <name>INTLEVEL0</name>
+              <description>INTLEVEL0</description>
               <bitOffset>0</bitOffset>
               <bitWidth>1</bitWidth>
               <access>read-write</access>
             </field>
             <field>
-              <name>CPMOS</name>
-              <description>CPMOS</description>
+              <name>INTLEVEL1</name>
+              <description>INTLEVEL1</description>
               <bitOffset>1</bitOffset>
               <bitWidth>1</bitWidth>
               <access>read-write</access>
             </field>
             <field>
-              <name>CPCEL</name>
-              <description>CPCEL</description>
+              <name>INTLEVEL2</name>
+              <description>INTLEVEL2</description>
               <bitOffset>2</bitOffset>
               <bitWidth>1</bitWidth>
               <access>read-write</access>
             </field>
-          </fields>
-        </register>
-        <register>
-          <name>CPINTOF</name>
-          <description>CPINTOF</description>
-          <addressOffset>0x0000004C</addressOffset>
-          <access>read-write</access>
-          <resetValue>0x00000000</resetValue>
-          <resetMask>0xFFFFFFFF</resetMask>
-          <fields>
             <field>
-              <name>CPRAW</name>
-              <description>CPRAW</description>
-              <bitOffset>0</bitOffset>
+              <name>INTLEVEL3</name>
+              <description>INTLEVEL3</description>
+              <bitOffset>3</bitOffset>
               <bitWidth>1</bitWidth>
               <access>read-write</access>
             </field>
             <field>
-              <name>CPMOS</name>
-              <description>CPMOS</description>
-              <bitOffset>1</bitOffset>
+              <name>INTLEVEL4</name>
+              <description>INTLEVEL4</description>
+              <bitOffset>4</bitOffset>
               <bitWidth>1</bitWidth>
               <access>read-write</access>
             </field>
             <field>
-              <name>CPCEL</name>
-              <description>CPCEL</description>
-              <bitOffset>2</bitOffset>
+              <name>INTLEVEL5</name>
+              <description>INTLEVEL5</description>
+              <bitOffset>5</bitOffset>
               <bitWidth>1</bitWidth>
               <access>read-write</access>
             </field>
-          </fields>
-        </register>
-        <register>
-          <name>CPINTCLR</name>
-          <description>CPINTCLR</description>
-          <addressOffset>0x00000050</addressOffset>
-          <access>read-write</access>
-          <resetValue>0x00000000</resetValue>
-          <resetMask>0xFFFFFFFF</resetMask>
-          <fields>
             <field>
-              <name>CPRAW</name>
-              <description>CPRAW</description>
-              <bitOffset>0</bitOffset>
+              <name>INTLEVEL6</name>
+              <description>INTLEVEL6</description>
+              <bitOffset>6</bitOffset>
               <bitWidth>1</bitWidth>
               <access>read-write</access>
             </field>
             <field>
-              <name>CPMOS</name>
-              <description>CPMOS</description>
-              <bitOffset>1</bitOffset>
+              <name>INTLEVEL7</name>
+              <description>INTLEVEL7</description>
+              <bitOffset>7</bitOffset>
               <bitWidth>1</bitWidth>
               <access>read-write</access>
             </field>
             <field>
-              <name>CPCEL</name>
-              <description>CPCEL</description>
-              <bitOffset>2</bitOffset>
+              <name>INTLEVEL8</name>
+              <description>INTLEVEL8</description>
+              <bitOffset>8</bitOffset>
+              <bitWidth>1</bitWidth>
+              <access>read-write</access>
+            </field>
+            <field>
+              <name>INTLEVEL9</name>
+              <description>INTLEVEL9</description>
+              <bitOffset>9</bitOffset>
               <bitWidth>1</bitWidth>
               <access>read-write</access>
             </field>
           </fields>
         </register>
-      </registers>
-    </peripheral>
-    <peripheral>
-      <name>EXTIA</name>
-      <description>Registers group</description>
-      <groupName>EXTIA</groupName>
-      <baseAddress>0x40001030</baseAddress>
-      <addressBlock>
-        <offset>0x00000000</offset>
-        <size>0x00000800</size>
-        <usage>registers</usage>
-      </addressBlock>
-      <registers>
         <register>
-          <name>INTEN</name>
-          <description>INTEN</description>
-          <addressOffset>0x00000000</addressOffset>
-          <access>read-write</access>
+          <name>INTSTAT</name>
+          <description>INTSTAT</description>
+          <addressOffset>0x00000040</addressOffset>
+          <access>read-only</access>
           <resetValue>0x00000000</resetValue>
           <resetMask>0xFFFFFFFF</resetMask>
           <fields>
             <field>
-              <name>INTEN0</name>
-              <description>INTEN0</description>
+              <name>INTSTAT0</name>
+              <description>INTSTAT0</description>
               <bitOffset>0</bitOffset>
               <bitWidth>1</bitWidth>
               <access>read-write</access>
             </field>
             <field>
-              <name>INTEN1</name>
-              <description>INTEN1</description>
+              <name>INTSTAT1</name>
+              <description>INTSTAT1</description>
               <bitOffset>1</bitOffset>
               <bitWidth>1</bitWidth>
               <access>read-write</access>
             </field>
             <field>
-              <name>INTEN2</name>
-              <description>INTEN2</description>
+              <name>INTSTAT2</name>
+              <description>INTSTAT2</description>
               <bitOffset>2</bitOffset>
               <bitWidth>1</bitWidth>
               <access>read-write</access>
             </field>
             <field>
-              <name>INTEN3</name>
-              <description>INTEN3</description>
+              <name>INTSTAT3</name>
+              <description>INTSTAT3</description>
               <bitOffset>3</bitOffset>
               <bitWidth>1</bitWidth>
               <access>read-write</access>
             </field>
             <field>
-              <name>INTEN4</name>
-              <description>INTEN4</description>
+              <name>INTSTAT4</name>
+              <description>INTSTAT4</description>
               <bitOffset>4</bitOffset>
               <bitWidth>1</bitWidth>
               <access>read-write</access>
             </field>
             <field>
-              <name>INTEN5</name>
-              <description>INTEN5</description>
+              <name>INTSTAT5</name>
+              <description>INTSTAT5</description>
               <bitOffset>5</bitOffset>
               <bitWidth>1</bitWidth>
               <access>read-write</access>
             </field>
             <field>
-              <name>INTEN6</name>
-              <description>INTEN6</description>
+              <name>INTSTAT6</name>
+              <description>INTSTAT6</description>
               <bitOffset>6</bitOffset>
               <bitWidth>1</bitWidth>
               <access>read-write</access>
             </field>
             <field>
-              <name>INTEN7</name>
-              <description>INTEN7</description>
+              <name>INTSTAT7</name>
+              <description>INTSTAT7</description>
               <bitOffset>7</bitOffset>
               <bitWidth>1</bitWidth>
               <access>read-write</access>
             </field>
             <field>
-              <name>INTEN8</name>
-              <description>INTEN8</description>
+              <name>INTSTAT8</name>
+              <description>INTSTAT8</description>
               <bitOffset>8</bitOffset>
               <bitWidth>1</bitWidth>
               <access>read-write</access>
             </field>
             <field>
-              <name>INTEN9</name>
-              <description>INTEN9</description>
+              <name>INTSTAT9</name>
+              <description>INTSTAT9</description>
               <bitOffset>9</bitOffset>
               <bitWidth>1</bitWidth>
               <access>read-write</access>
@@ -1139,79 +968,79 @@
           </fields>
         </register>
         <register>
-          <name>INTMSK</name>
-          <description>INTMSK</description>
-          <addressOffset>0x00000004</addressOffset>
-          <access>read-write</access>
+          <name>INTRAWS</name>
+          <description>INTRAWS</description>
+          <addressOffset>0x00000044</addressOffset>
+          <access>read-only</access>
           <resetValue>0x00000000</resetValue>
           <resetMask>0xFFFFFFFF</resetMask>
           <fields>
             <field>
-              <name>INTMSK0</name>
-              <description>INTMSK0</description>
+              <name>INTRAWS0</name>
+              <description>INTRAWS0</description>
               <bitOffset>0</bitOffset>
               <bitWidth>1</bitWidth>
               <access>read-write</access>
             </field>
             <field>
-              <name>INTMSK1</name>
-              <description>INTMSK1</description>
+              <name>INTRAWS1</name>
+              <description>INTRAWS1</description>
               <bitOffset>1</bitOffset>
               <bitWidth>1</bitWidth>
               <access>read-write</access>
             </field>
             <field>
-              <name>INTMSK2</name>
-              <description>INTMSK2</description>
+              <name>INTRAWS2</name>
+              <description>INTRAWS2</description>
               <bitOffset>2</bitOffset>
               <bitWidth>1</bitWidth>
               <access>read-write</access>
             </field>
             <field>
-              <name>INTMSK3</name>
-              <description>INTMSK3</description>
+              <name>INTRAWS3</name>
+              <description>INTRAWS3</description>
               <bitOffset>3</bitOffset>
               <bitWidth>1</bitWidth>
               <access>read-write</access>
             </field>
             <field>
-              <name>INTMSK4</name>
-              <description>INTMSK4</description>
+              <name>INTRAWS4</name>
+              <description>INTRAWS4</description>
               <bitOffset>4</bitOffset>
               <bitWidth>1</bitWidth>
               <access>read-write</access>
             </field>
             <field>
-              <name>INTMSK5</name>
-              <description>INTMSK5</description>
+              <name>INTRAWS5</name>
+              <description>INTRAWS5</description>
               <bitOffset>5</bitOffset>
               <bitWidth>1</bitWidth>
               <access>read-write</access>
             </field>
             <field>
-              <name>INTMSK6</name>
-              <description>INTMSK6</description>
+              <name>INTRAWS6</name>
+              <description>INTRAWS6</description>
               <bitOffset>6</bitOffset>
               <bitWidth>1</bitWidth>
               <access>read-write</access>
             </field>
             <field>
-              <name>INTMSK7</name>
-              <description>INTMSK7</description>
+              <name>INTRAWS7</name>
+              <description>INTRAWS7</description>
               <bitOffset>7</bitOffset>
               <bitWidth>1</bitWidth>
               <access>read-write</access>
             </field>
             <field>
-              <name>INTMSK8</name>
-              <description>INTMSK8</description>
+              <name>INTRAWS8</name>
+              <description>INTRAWS8</description>
               <bitOffset>8</bitOffset>
               <bitWidth>1</bitWidth>
               <access>read-write</access>
             </field>
             <field>
-              <name>INTMSK9</name>
-              <description>INTMSK9</description>
+              <name>INTRAWS9</name>
+              <description>INTRAWS9</description>
               <bitOffset>9</bitOffset>
               <bitWidth>1</bitWidth>
               <access>read-write</access>
@@ -1219,79 +1048,79 @@
           </fields>
         </register>
         <register>
-          <name>INTMODE</name>
-          <description>INTMODE</description>
-          <addressOffset>0x00000008</addressOffset>
+          <name>INTCLR</name>
+          <description>INTCLR</description>
+          <addressOffset>0x0000004C</addressOffset>
           <access>read-write</access>
           <resetValue>0x00000000</resetValue>
           <resetMask>0xFFFFFFFF</resetMask>
           <fields>
             <field>
-              <name>INTMODE0</name>
-              <description>INTMODE0</description>
+              <name>INTCLR0</name>
+              <description>INTCLR0</description>
               <bitOffset>0</bitOffset>
               <bitWidth>1</bitWidth>
               <access>read-write</access>
             </field>
             <field>
-              <name>INTMODE1</name>
-              <description>INTMODE1</description>
+              <name>INTCLR1</name>
+              <description>INTCLR1</description>
               <bitOffset>1</bitOffset>
               <bitWidth>1</bitWidth>
               <access>read-write</access>
             </field>
             <field>
-              <name>INTMODE2</name>
-              <description>INTMODE2</description>
+              <name>INTCLR2</name>
+              <description>INTCLR2</description>
               <bitOffset>2</bitOffset>
               <bitWidth>1</bitWidth>
               <access>read-write</access>
             </field>
             <field>
-              <name>INTMODE3</name>
-              <description>INTMODE3</description>
+              <name>INTCLR3</name>
+              <description>INTCLR3</description>
               <bitOffset>3</bitOffset>
               <bitWidth>1</bitWidth>
               <access>read-write</access>
             </field>
             <field>
-              <name>INTMODE4</name>
-              <description>INTMODE4</description>
+              <name>INTCLR4</name>
+              <description>INTCLR4</description>
               <bitOffset>4</bitOffset>
               <bitWidth>1</bitWidth>
               <access>read-write</access>
             </field>
             <field>
-              <name>INTMODE5</name>
-              <description>INTMODE5</description>
+              <name>INTCLR5</name>
+              <description>INTCLR5</description>
               <bitOffset>5</bitOffset>
               <bitWidth>1</bitWidth>
               <access>read-write</access>
             </field>
             <field>
-              <name>INTMODE6</name>
-              <description>INTMODE6</description>
+              <name>INTCLR6</name>
+              <description>INTCLR6</description>
               <bitOffset>6</bitOffset>
               <bitWidth>1</bitWidth>
               <access>read-write</access>
             </field>
             <field>
-              <name>INTMODE7</name>
-              <description>INTMODE7</description>
+              <name>INTCLR7</name>
+              <description>INTCLR7</description>
               <bitOffset>7</bitOffset>
               <bitWidth>1</bitWidth>
               <access>read-write</access>
             </field>
             <field>
-              <name>INTMODE8</name>
-              <description>INTMODE8</description>
+              <name>INTCLR8</name>
+              <description>INTCLR8</description>
               <bitOffset>8</bitOffset>
               <bitWidth>1</bitWidth>
               <access>read-write</access>
             </field>
             <field>
-              <name>INTMODE9</name>
-              <description>INTMODE9</description>
+              <name>INTCLR9</name>
+              <description>INTCLR9</description>
               <bitOffset>9</bitOffset>
               <bitWidth>1</bitWidth>
               <access>read-write</access>
@@ -1299,320 +1128,465 @@
           </fields>
         </register>
         <register>
-          <name>INTLEVEL</name>
-          <description>INTLEVEL</description>
-          <addressOffset>0x0000000C</addressOffset>
-          <access>read-write</access>
+          <name>EXT</name>
+          <description>EXT</description>
+          <addressOffset>0x00000050</addressOffset>
+          <access>read-only</access>
           <resetValue>0x00000000</resetValue>
           <resetMask>0xFFFFFFFF</resetMask>
           <fields>
             <field>
-              <name>INTLEVEL0</name>
-              <description>INTLEVEL0</description>
+              <name>EXT0</name>
+              <description>EXT0</description>
               <bitOffset>0</bitOffset>
               <bitWidth>1</bitWidth>
               <access>read-write</access>
             </field>
             <field>
-              <name>INTLEVEL1</name>
-              <description>INTLEVEL1</description>
+              <name>EXT1</name>
+              <description>EXT1</description>
               <bitOffset>1</bitOffset>
               <bitWidth>1</bitWidth>
               <access>read-write</access>
             </field>
             <field>
-              <name>INTLEVEL2</name>
-              <description>INTLEVEL2</description>
+              <name>EXT2</name>
+              <description>EXT2</description>
               <bitOffset>2</bitOffset>
               <bitWidth>1</bitWidth>
               <access>read-write</access>
             </field>
             <field>
-              <name>INTLEVEL3</name>
-              <description>INTLEVEL3</description>
+              <name>EXT3</name>
+              <description>EXT3</description>
               <bitOffset>3</bitOffset>
               <bitWidth>1</bitWidth>
               <access>read-write</access>
             </field>
             <field>
-              <name>INTLEVEL4</name>
-              <description>INTLEVEL4</description>
+              <name>EXT4</name>
+              <description>EXT4</description>
               <bitOffset>4</bitOffset>
               <bitWidth>1</bitWidth>
               <access>read-write</access>
             </field>
             <field>
-              <name>INTLEVEL5</name>
-              <description>INTLEVEL5</description>
+              <name>EXT5</name>
+              <description>EXT5</description>
               <bitOffset>5</bitOffset>
               <bitWidth>1</bitWidth>
               <access>read-write</access>
             </field>
             <field>
-              <name>INTLEVEL6</name>
-              <description>INTLEVEL6</description>
+              <name>EXT6</name>
+              <description>EXT6</description>
               <bitOffset>6</bitOffset>
               <bitWidth>1</bitWidth>
               <access>read-write</access>
             </field>
             <field>
-              <name>INTLEVEL7</name>
-              <description>INTLEVEL7</description>
+              <name>EXT7</name>
+              <description>EXT7</description>
               <bitOffset>7</bitOffset>
               <bitWidth>1</bitWidth>
               <access>read-write</access>
             </field>
             <field>
-              <name>INTLEVEL8</name>
-              <description>INTLEVEL8</description>
+              <name>EXT8</name>
+              <description>EXT8</description>
               <bitOffset>8</bitOffset>
               <bitWidth>1</bitWidth>
               <access>read-write</access>
             </field>
             <field>
-              <name>INTLEVEL9</name>
-              <description>INTLEVEL9</description>
+              <name>EXT9</name>
+              <description>EXT9</description>
               <bitOffset>9</bitOffset>
               <bitWidth>1</bitWidth>
               <access>read-write</access>
             </field>
           </fields>
         </register>
+      </registers>
+    </peripheral>
+    <peripheral>
+      <name>ACMP</name>
+      <description>Registers group</description>
+      <groupName>ACMP</groupName>
+      <baseAddress>0x40000800</baseAddress>
+      <addressBlock>
+        <offset>0x00000000</offset>
+        <size>0x00000800</size>
+        <usage>registers</usage>
+      </addressBlock>
+      <registers>
         <register>
-          <name>INTSTAT</name>
-          <description>INTSTAT</description>
+          <name>CPEN</name>
+          <description>CPEN</description>
+          <addressOffset>0x00000000</addressOffset>
+          <access>read-write</access>
+          <resetValue>0x00000000</resetValue>
+          <resetMask>0xFFFFFFFF</resetMask>
+          <fields>
+            <field>
+              <name>EN</name>
+              <description>EN</description>
+              <bitOffset>4</bitOffset>
+              <bitWidth>1</bitWidth>
+              <access>read-write</access>
+            </field>
+          </fields>
+        </register>
+        <register>
+          <name>CPLEVEL</name>
+          <description>CPLEVEL</description>
+          <addressOffset>0x00000004</addressOffset>
+          <access>read-write</access>
+          <resetValue>0x00000000</resetValue>
+          <resetMask>0xFFFFFFFF</resetMask>
+          <fields>
+            <field>
+              <name>LEVEL</name>
+              <description>LEVEL</description>
+              <bitOffset>0</bitOffset>
+              <bitWidth>8</bitWidth>
+              <access>read-write</access>
+            </field>
+          </fields>
+        </register>
+        <register>
+          <name>CPOUTPUT</name>
+          <description>CPOUTPUT</description>
           <addressOffset>0x00000010</addressOffset>
-          <access>read-only</access>
+          <access>read-write</access>
           <resetValue>0x00000000</resetValue>
           <resetMask>0xFFFFFFFF</resetMask>
           <fields>
             <field>
-              <name>INTSTAT0</name>
-              <description>INTSTAT0</description>
+              <name>CPOUTRAW</name>
+              <description>CPOUTRAW</description>
               <bitOffset>0</bitOffset>
               <bitWidth>1</bitWidth>
               <access>read-write</access>
             </field>
             <field>
-              <name>INTSTAT1</name>
-              <description>INTSTAT1</description>
+              <name>CPOUTMOS</name>
+              <description>CPOUTMOS</description>
               <bitOffset>1</bitOffset>
               <bitWidth>1</bitWidth>
               <access>read-write</access>
             </field>
             <field>
-              <name>INTSTAT2</name>
-              <description>INTSTAT2</description>
+              <name>CPOUTCEL</name>
+              <description>CPOUTCEL</description>
               <bitOffset>2</bitOffset>
               <bitWidth>1</bitWidth>
               <access>read-write</access>
             </field>
+          </fields>
+        </register>
+        <register>
+          <name>CPMOD</name>
+          <description>CPMOD</description>
+          <addressOffset>0x00000020</addressOffset>
+          <access>read-write</access>
+          <resetValue>0x00000000</resetValue>
+          <resetMask>0xFFFFFFFF</resetMask>
+          <fields>
             <field>
-              <name>INTSTAT3</name>
-              <description>INTSTAT3</description>
-              <bitOffset>3</bitOffset>
+              <name>WORKSMOD</name>
+              <description>WORKSMOD</description>
+              <bitOffset>0</bitOffset>
               <bitWidth>1</bitWidth>
               <access>read-write</access>
             </field>
+          </fields>
+        </register>
+        <register>
+          <name>CPCTL</name>
+          <description>CPCTL</description>
+          <addressOffset>0x00000024</addressOffset>
+          <access>read-write</access>
+          <resetValue>0x00000000</resetValue>
+          <resetMask>0xFFFFFFFF</resetMask>
+          <fields>
             <field>
-              <name>INTSTAT4</name>
-              <description>INTSTAT4</description>
-              <bitOffset>4</bitOffset>
+              <name>MOS_LEVEL</name>
+              <description>MOS_LEVEL</description>
+              <bitOffset>1</bitOffset>
               <bitWidth>1</bitWidth>
               <access>read-write</access>
             </field>
             <field>
-              <name>INTSTAT5</name>
-              <description>INTSTAT5</description>
-              <bitOffset>5</bitOffset>
+              <name>CEL_LEVEL</name>
+              <description>CEL_LEVEL</description>
+              <bitOffset>2</bitOffset>
               <bitWidth>1</bitWidth>
               <access>read-write</access>
             </field>
             <field>
-              <name>INTSTAT6</name>
-              <description>INTSTAT6</description>
-              <bitOffset>6</bitOffset>
+              <name>A7M_LEVEL</name>
+              <description>A7M_LEVEL</description>
+              <bitOffset>3</bitOffset>
               <bitWidth>1</bitWidth>
               <access>read-write</access>
             </field>
             <field>
-              <name>INTSTAT7</name>
-              <description>INTSTAT7</description>
-              <bitOffset>7</bitOffset>
+              <name>A7C_LEVEL</name>
+              <description>A7C_LEVEL</description>
+              <bitOffset>4</bitOffset>
               <bitWidth>1</bitWidth>
               <access>read-write</access>
             </field>
             <field>
-              <name>INTSTAT8</name>
-              <description>INTSTAT8</description>
+              <name>MOS_DELAY</name>
+              <description>MOS_DELAY</description>
               <bitOffset>8</bitOffset>
-              <bitWidth>1</bitWidth>
+              <bitWidth>6</bitWidth>
               <access>read-write</access>
             </field>
             <field>
-              <name>INTSTAT9</name>
-              <description>INTSTAT9</description>
-              <bitOffset>9</bitOffset>
+              <name>CEL_DELAY</name>
+              <description>CEL_DELAY</description>
+              <bitOffset>16</bitOffset>
+              <bitWidth>6</bitWidth>
+              <access>read-write</access>
+            </field>
+          </fields>
+        </register>
+        <register>
+          <name>CLRLOCK</name>
+          <description>CLRLOCK</description>
+          <addressOffset>0x00000028</addressOffset>
+          <access>read-write</access>
+          <resetValue>0x00000000</resetValue>
+          <resetMask>0xFFFFFFFF</resetMask>
+          <fields>
+            <field>
+              <name>CLR</name>
+              <description>CLR</description>
+              <bitOffset>0</bitOffset>
               <bitWidth>1</bitWidth>
               <access>read-write</access>
             </field>
           </fields>
         </register>
         <register>
-          <name>INTRAWS</name>
-          <description>INTRAWS</description>
-          <addressOffset>0x00000014</addressOffset>
-          <access>read-only</access>
+          <name>LOCKCON</name>
+          <description>LOCKCON</description>
+          <addressOffset>0x0000002C</addressOffset>
+          <access>read-write</access>
           <resetValue>0x00000000</resetValue>
           <resetMask>0xFFFFFFFF</resetMask>
           <fields>
             <field>
-              <name>INTRAWS0</name>
-              <description>INTRAWS0</description>
+              <name>LOCKEN</name>
+              <description>LOCKEN</description>
               <bitOffset>0</bitOffset>
               <bitWidth>1</bitWidth>
               <access>read-write</access>
             </field>
             <field>
-              <name>INTRAWS1</name>
-              <description>INTRAWS1</description>
+              <name>LEVELCON</name>
+              <description>LEVELCON</description>
               <bitOffset>1</bitOffset>
               <bitWidth>1</bitWidth>
               <access>read-write</access>
             </field>
             <field>
-              <name>INTRAWS2</name>
-              <description>INTRAWS2</description>
+              <name>CLRCON</name>
+              <description>CLRCON</description>
               <bitOffset>2</bitOffset>
               <bitWidth>1</bitWidth>
               <access>read-write</access>
             </field>
             <field>
-              <name>INTRAWS3</name>
-              <description>INTRAWS3</description>
+              <name>LOCKSTAT</name>
+              <description>LOCKSTAT</description>
               <bitOffset>3</bitOffset>
               <bitWidth>1</bitWidth>
               <access>read-write</access>
             </field>
+          </fields>
+        </register>
+        <register>
+          <name>CPINTCON</name>
+          <description>CPINTCON</description>
+          <addressOffset>0x00000040</addressOffset>
+          <access>read-write</access>
+          <resetValue>0x00000000</resetValue>
+          <resetMask>0xFFFFFFFF</resetMask>
+          <fields>
             <field>
-              <name>INTRAWS4</name>
-              <description>INTRAWS4</description>
-              <bitOffset>4</bitOffset>
+              <name>INTMASKn</name>
+              <description>INTMASKn</description>
+              <bitOffset>0</bitOffset>
               <bitWidth>1</bitWidth>
               <access>read-write</access>
             </field>
             <field>
-              <name>INTRAWS5</name>
-              <description>INTRAWS5</description>
-              <bitOffset>5</bitOffset>
+              <name>INTUP</name>
+              <description>INTUP</description>
+              <bitOffset>2</bitOffset>
               <bitWidth>1</bitWidth>
               <access>read-write</access>
             </field>
             <field>
-              <name>INTRAWS6</name>
-              <description>INTRAWS6</description>
-              <bitOffset>6</bitOffset>
+              <name>INTDOWN</name>
+              <description>INTDOWN</description>
+              <bitOffset>3</bitOffset>
               <bitWidth>1</bitWidth>
               <access>read-write</access>
             </field>
             <field>
-              <name>INTRAWS7</name>
-              <description>INTRAWS7</description>
-              <bitOffset>7</bitOffset>
+              <name>MOSINTMASKn</name>
+              <description>MOSINTMASKn</description>
+              <bitOffset>4</bitOffset>
               <bitWidth>1</bitWidth>
               <access>read-write</access>
             </field>
             <field>
-              <name>INTRAWS8</name>
-              <description>INTRAWS8</description>
-              <bitOffset>8</bitOffset>
+              <name>MOSINTEN</name>
+              <description>MOSINTEN</description>
+              <bitOffset>5</bitOffset>
               <bitWidth>1</bitWidth>
               <access>read-write</access>
             </field>
             <field>
-              <name>INTRAWS9</name>
-              <description>INTRAWS9</description>
-              <bitOffset>9</bitOffset>
+              <name>MOSINTUP</name>
+              <description>MOSINTUP</description>
+              <bitOffset>6</bitOffset>
+              <bitWidth>1</bitWidth>
+              <access>read-write</access>
+            </field>
+            <field>
+              <name>MOSINTDOWN</name>
+              <description>MOSINTDOWN</description>
+              <bitOffset>7</bitOffset>
               <bitWidth>1</bitWidth>
               <access>read-write</access>
             </field>
           </fields>
         </register>
         <register>
-          <name>INTCLR</name>
-          <description>INTCLR</description>
-          <addressOffset>0x0000001C</addressOffset>
+          <name>CPINTRAW</name>
+          <description>CPINTRAW</description>
+          <addressOffset>0x00000044</addressOffset>
           <access>read-write</access>
           <resetValue>0x00000000</resetValue>
           <resetMask>0xFFFFFFFF</resetMask>
           <fields>
             <field>
-              <name>INTCLR0</name>
-              <description>INTCLR0</description>
+              <name>CPRAW</name>
+              <description>CPRAW</description>
               <bitOffset>0</bitOffset>
               <bitWidth>1</bitWidth>
               <access>read-write</access>
             </field>
             <field>
-              <name>INTCLR1</name>
-              <description>INTCLR1</description>
+              <name>CPMOS</name>
+              <description>CPMOS</description>
               <bitOffset>1</bitOffset>
               <bitWidth>1</bitWidth>
               <access>read-write</access>
             </field>
             <field>
-              <name>INTCLR2</name>
-              <description>INTCLR2</description>
+              <name>CPCEL</name>
+              <description>CPCEL</description>
               <bitOffset>2</bitOffset>
               <bitWidth>1</bitWidth>
               <access>read-write</access>
             </field>
+          </fields>
+        </register>
+        <register>
+          <name>CPINT</name>
+          <description>CPINT</description>
+          <addressOffset>0x00000048</addressOffset>
+          <access>read-write</access>
+          <resetValue>0x00000000</resetValue>
+          <resetMask>0xFFFFFFFF</resetMask>
+          <fields>
             <field>
-              <name>INTCLR3</name>
-              <description>INTCLR3</description>
-              <bitOffset>3</bitOffset>
+              <name>CPRAW</name>
+              <description>CPRAW</description>
+              <bitOffset>0</bitOffset>
               <bitWidth>1</bitWidth>
               <access>read-write</access>
             </field>
             <field>
-              <name>INTCLR4</name>
-              <description>INTCLR4</description>
-              <bitOffset>4</bitOffset>
+              <name>CPMOS</name>
+              <description>CPMOS</description>
+              <bitOffset>1</bitOffset>
               <bitWidth>1</bitWidth>
               <access>read-write</access>
             </field>
             <field>
-              <name>INTCLR5</name>
-              <description>INTCLR5</description>
-              <bitOffset>5</bitOffset>
+              <name>CPCEL</name>
+              <description>CPCEL</description>
+              <bitOffset>2</bitOffset>
+              <bitWidth>1</bitWidth>
+              <access>read-write</access>
+            </field>
+          </fields>
+        </register>
+        <register>
+          <name>CPINTOF</name>
+          <description>CPINTOF</description>
+          <addressOffset>0x0000004C</addressOffset>
+          <access>read-write</access>
+          <resetValue>0x00000000</resetValue>
+          <resetMask>0xFFFFFFFF</resetMask>
+          <fields>
+            <field>
+              <name>CPRAW</name>
+              <description>CPRAW</description>
+              <bitOffset>0</bitOffset>
               <bitWidth>1</bitWidth>
               <access>read-write</access>
             </field>
             <field>
-              <name>INTCLR6</name>
-              <description>INTCLR6</description>
-              <bitOffset>6</bitOffset>
+              <name>CPMOS</name>
+              <description>CPMOS</description>
+              <bitOffset>1</bitOffset>
               <bitWidth>1</bitWidth>
               <access>read-write</access>
             </field>
             <field>
-              <name>INTCLR7</name>
-              <description>INTCLR7</description>
-              <bitOffset>7</bitOffset>
+              <name>CPCEL</name>
+              <description>CPCEL</description>
+              <bitOffset>2</bitOffset>
+              <bitWidth>1</bitWidth>
+              <access>read-write</access>
+            </field>
+          </fields>
+        </register>
+        <register>
+          <name>CPINTCLR</name>
+          <description>CPINTCLR</description>
+          <addressOffset>0x00000050</addressOffset>
+          <access>read-write</access>
+          <resetValue>0x00000000</resetValue>
+          <resetMask>0xFFFFFFFF</resetMask>
+          <fields>
+            <field>
+              <name>CPRAW</name>
+              <description>CPRAW</description>
+              <bitOffset>0</bitOffset>
               <bitWidth>1</bitWidth>
               <access>read-write</access>
             </field>
             <field>
-              <name>INTCLR8</name>
-              <description>INTCLR8</description>
-              <bitOffset>8</bitOffset>
+              <name>CPMOS</name>
+              <description>CPMOS</description>
+              <bitOffset>1</bitOffset>
               <bitWidth>1</bitWidth>
               <access>read-write</access>
             </field>
             <field>
-              <name>INTCLR9</name>
-              <description>INTCLR9</description>
-              <bitOffset>9</bitOffset>
+              <name>CPCEL</name>
+              <description>CPCEL</description>
+              <bitOffset>2</bitOffset>
               <bitWidth>1</bitWidth>
               <access>read-write</access>
             </field>
