--- SWM050-patched.svd.orig	2019-04-08 10:29:43.639088116 +0200
+++ SWM050-patched.svd	2019-04-08 10:46:36.766538447 +0200
@@ -31,14 +31,14 @@
           <fields>
             <field>
               <name>CORE_CLK_DIV</name>
-              <description>=0：2分频			=奇数：不分频		=偶数 分频</description>
+              <description>=0:2 Divide = Odd: Undivided = Even Divided, directliy divides the clock source, independent of the core clock</description>
               <bitOffset>0</bitOffset>
               <bitWidth>10</bitWidth>
               <access>read-write</access>
             </field>
             <field>
               <name>TMRSE_CLK_DIV</name>
-              <description>=0：2分频			=奇数：不分频		=偶数 分频，直接对时钟源分频，不依赖于内核时钟</description>
+              <description>=0:2 Divide = Odd: Divide = Even Divided</description>
               <bitOffset>16</bitOffset>
               <bitWidth>6</bitWidth>
               <access>read-write</access>
@@ -86,7 +86,7 @@
           <fields>
             <field>
               <name>DBL_IRC</name>
-              <description>0 18MHz	   1 36MHz</description>
+              <description>0:18MHz	   1:36MHz</description>
               <bitOffset>0</bitOffset>
               <bitWidth>1</bitWidth>
               <access>read-write</access>
@@ -147,7 +147,7 @@
           <fields>
             <field>
               <name>EN</name>
-              <description>0 A1/A2功能由PORT_SEL决定		1 A1/A2连接到SWD功能，分别为SWDIO/SWCLK功能</description>
+              <description>0 A1/A2 function determined by PORT_SEL 1 A1/A2 connected to SWD function, respectively SWDIO/SWCLK function</description>
               <bitOffset>0</bitOffset>
               <bitWidth>1</bitWidth>
               <access>read-write</access>
@@ -244,7 +244,7 @@
           <fields>
             <field>
               <name>PA00</name>
-              <description>0 上拉关闭		1 上拉开启</description>
+              <description>0 pull up closed 1 pull up open</description>
               <bitOffset>0</bitOffset>
               <bitWidth>1</bitWidth>
               <access>read-write</access>
@@ -324,7 +324,7 @@
           <fields>
             <field>
               <name>PA00</name>
-              <description>0 GPIOA.0输入功能不禁止	1 GPIOA.0输入功能禁止，以降低功耗</description>
+              <description>0 GPIOA.0 input function is not disabled 1 GPIOA.0 input function is disabled to reduce power consumption</description>
               <bitOffset>0</bitOffset>
               <bitWidth>1</bitWidth>
               <access>read-write</access>
@@ -1622,7 +1622,7 @@
             </field>
             <field>
               <name>RMOD</name>
-              <description>0 超时即复位		1 第一次超时产生中断，若产生第二次超时时，没有清除中断，则系统复位</description>
+              <description>0 Timeout is reset 1 The first time-out generates an interrupt. If a second time-out occurs, the interrupt is not cleared. The system resets.</description>
               <bitOffset>1</bitOffset>
               <bitWidth>1</bitWidth>
               <access>read-write</access>
@@ -1639,14 +1639,14 @@
           <fields>
             <field>
               <name>TOP</name>
-              <description>初始值（在WDT使能之前写入值）	2^(8+TOP)，即24位</description>
+              <description>Initial value (write value before WDT enable) 2^(8+TOP), 24 bits</description>
               <bitOffset>0</bitOffset>
               <bitWidth>4</bitWidth>
               <access>read-write</access>
             </field>
             <field>
               <name>TOP_INIT</name>
-              <description>超时后将要填装的值</description>
+              <description>The value to be filled after timeout</description>
               <bitOffset>4</bitOffset>
               <bitWidth>4</bitWidth>
               <access>read-write</access>
@@ -1712,49 +1712,49 @@
           <fields>
             <field>
               <name>ENA</name>
-              <description>模块使能</description>
+              <description>Enable Timer</description>
               <bitOffset>0</bitOffset>
               <bitWidth>1</bitWidth>
               <access>read-write</access>
             </field>
             <field>
               <name>WMOD</name>
-              <description>00 定时器/计数器模式	01 PWM模式	10 脉宽捕捉模式		11 占空比捕捉模式</description>
+              <description>00 Timer/Counter Mode 01 PWM Mode 10 Pulse Width Capture Mode 11 Duty Cycle Capture Mode</description>
               <bitOffset>4</bitOffset>
               <bitWidth>2</bitWidth>
               <access>read-write</access>
             </field>
             <field>
               <name>OSCMOD</name>
-              <description>0 内部	1 外部</description>
+              <description>0 Internal 1 External</description>
               <bitOffset>8</bitOffset>
               <bitWidth>1</bitWidth>
               <access>read-write</access>
             </field>
             <field>
               <name>OUTMOD</name>
-              <description>00 无输出		01 取反		10 置高		11 置低</description>
+              <description>00 No output 01 Invert 10 Set high 11 Set low</description>
               <bitOffset>12</bitOffset>
               <bitWidth>2</bitWidth>
               <access>read-write</access>
             </field>
             <field>
               <name>EDGE_F</name>
-              <description>0 上升沿		1 下降沿</description>
+              <description>0 Rising edge 1 Falling edge</description>
               <bitOffset>16</bitOffset>
               <bitWidth>1</bitWidth>
               <access>read-write</access>
             </field>
             <field>
               <name>VALSAVE</name>
-              <description>保留计数值</description>
+              <description>Keep count value</description>
               <bitOffset>24</bitOffset>
               <bitWidth>1</bitWidth>
               <access>read-write</access>
             </field>
             <field>
               <name>SINGLE</name>
-              <description>1 单次模式 0 循环模式</description>
+              <description>1 single mode 0 circular mode</description>
               <bitOffset>28</bitOffset>
               <bitWidth>1</bitWidth>
               <access>read-write</access>
@@ -1825,14 +1825,14 @@
           <fields>
             <field>
               <name>ENA</name>
-              <description>1 使能	to observe sate, you should set this bit 1</description>
+              <description>1 Enable	to observe sate, you should set this bit 1</description>
               <bitOffset>0</bitOffset>
               <bitWidth>1</bitWidth>
               <access>read-write</access>
             </field>
             <field>
               <name>MASKn</name>
-              <description>0 屏蔽</description>
+              <description>0 shield</description>
               <bitOffset>1</bitOffset>
               <bitWidth>1</bitWidth>
               <access>read-write</access>
